### ISSUE TITLE
System-Cron errors/warnings still  appear after switching to Webcron or AJAX

### ISSUE THREAD

**Issue by steppy** [owncloud/core#18454]
Errors during system-cron execution are shown on top of the Administrator page ("Sicherheits- & Einrichtungswarnungen" in german). If the cron setting is set to webcron or ajax afterwards, these warning are not relevant anymore but still shown. This is very confusing.

Problem is also described here: https://forum.owncloud.org/viewtopic.php?f=21&t=29886

**Comment by MorrisJobke** [comment]
Because Webcron also suffer from PHP timeouts (because it is also handled by the webserver). the usual cron (as a shell script) is run without these constraints, because it uses command line PHP that doesn't need timeouts and therefore can run longer jobs (as for example the news app stuff).

Regarding the warning: This is most likely because there happened something with the cron run and on change the state of the failure wasn't cleared.

**Comment by MorrisJobke** [comment]
@steppy Can you provide the cron error that is printed? Best would be a screenshot too.

**Comment by steppy** [comment]
I finally managed to clear the warnings (which was quite an effort and included the cloud to be not working temporary). So I cannot provide a screenshot. But to make sure that you know what I am talking about, here is a Screenshot.

![scr](https://cloud.githubusercontent.com/assets/10523236/9382557/5e90fc94-4742-11e5-9c24-9414d13effe8.png)


In Addition to the Memory Cache Warning I had these messages:
```
Die Ausführung des Cron-Jobs über die Kommandozeile war nicht möglich. Die folgenden technischen Fehler sind dabei aufgetreten:	
Data directory (/www/htdocs/[...]/data) is invalid Please check that the data directory contains a file ".ocdata" in its root.
Cannot write into "config" directory This can usually be fixed by <a href="https://doc.owncloud.org/server/8.1/go.php?to=admin-dir_permissions" target="_blank">giving the webserver write access to the config directory</a>.
Cannot write into "apps" directory This can usually be fixed by <a href="https://doc.owncloud.org/server/8.1/go.php?to=admin-dir_permissions" target="_blank">giving the webserver write access to the apps directory</a> or disabling the appstore in the config file.
Data directory (/www/htdocs/[...]/data) not writable by ownCloud Permissions can usually be fixed by <a href="https://doc.owncloud.org/server/8.1/go.php?to=admin-dir_permissions" target="_blank">giving the webserver write access to the root directory</a>.
```

With switching back to AJAX or Webcron and after a successful cron execution I would have expected these messages to be gone. However, there was no way to make it disappear.

**Comment by MorrisJobke** [comment]
Thanks @steppy 

Fix is in #18456
